### PR TITLE
Definition Tweaks - see #5314

### PR DESCRIPTION
**Pull request by weaverryan**
| Q | A |
| --- | --- |
| Doc fix? | yes |
| New docs? | no |
| Applies to | all |
| Fixed tickets | n/a |

See #5314 - I am quite a novice with the Configuration component :).

### CODE REVIEW

File: components/config/definition.rst [hunk 1]
```diff
@@ -212,6 +212,9 @@ Before defining the children of an array node, you can provide options like:
 ``addDefaultsIfNotSet()``
     If any child nodes have default values, use them if explicit values
     haven't been provided.
+``normalizeKeys(false)``
+    If called (with ``false``), keys with dashes are *not* normaled to underscores.
```
Comment (stof) [inline components/config/definition.rst]: typo here: ``normalized``

File: components/config/definition.rst [hunk 2]
```diff
@@ -212,6 +212,9 @@ Before defining the children of an array node, you can provide options like:
...
+    It is recommended to use this to avoid this unnecessary normalization.
```
Comment (stof) [inline components/config/definition.rst]: only for prototyped nodes (and when it is a map, as it does not make sense for a list where keys are numeric)

File: components/config/definition.rst [hunk 3]
```diff
@@ -310,6 +313,12 @@ The output configuration will be exactly the same as before. In other words, the
 ``sf_connection`` and ``default`` configuration keys are lost. The reason is that
 the Symfony Config component treats arrays as lists by default.
 
+.. note::
+
+    As of writing this, there is an inconsistency: if only one file is processed,
```
Comment (stof) [inline components/config/definition.rst]: even when multiple files are processed (which happens in all Symfony projects, as nobody puts the whole configuration if ``config_prod`` or ``config_dev``), as long as only 1 file is providing config for this list

**Comment by weaverryan** [comment]
Thanks Stof - I needed the help on this one :)

**Pull request merged**